marshmallow_jsonschema/validation.py

```diff
@@ -100,3 +100,29 @@ def handle_range(schema, field, validator, parent_schema):
         schema['maximum'] = validator.max
 
     return schema
+
+
+def handle_regexp(schema, field, validator, parent_schema):
+    """Adds validation logic for ``marshmallow.validate.Regexp``, setting the
+    values appropriately for ``fields.String`` and its subclasses.
+
+    Args:
+        schema (dict): The original JSON schema we generated. This is what we
+            want to post-process.
+        field (fields.Field): The field that generated the original schema and
+            who this post-processor belongs to.
+        validator (marshmallow.validate.Regexp): The validator attached to the
+            passed in field.
+        parent_schema (marshmallow.Schema): The Schema instance that the field
+            belongs to.
+
+    Returns:
+        dict: A, possibly, new JSON Schema that has been post processed and
+            altered.
+    """
+    if not isinstance(field, fields.String):
```

Comment:
I recommend raising an `UnsupportedValueError` here as I did for other validators. marshmallow will fail on non-strings anyway:
```python
>>> from marshmallow import fields
>>> from marshmallow.validate import Regexp
>>> f = fields.Integer(validate=Regexp(".*"))
>>> f.deserialize(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lynx/projects/marshmallow-jsonschema/.venv/lib/python3.7/site-packages/marshmallow/fields.py", line 301, in deserialize
    self._validate(output)
  File "/home/lynx/projects/marshmallow-jsonschema/.venv/lib/python3.7/site-packages/marshmallow/fields.py", line 224, in _validate
    r = validator(value)
  File "/home/lynx/projects/marshmallow-jsonschema/.venv/lib/python3.7/site-packages/marshmallow/validate.py", line 362, in __call__
    if self.regex.match(value) is None:
TypeError: expected string or bytes-like object
```
Custom fields have to be handled with care though as they are not required to be an instance of `fields.String`. Though they are not supported yet, there is an attempt in #22.